### ISSUE TITLE
Property: "keyVaultSecretId" must use one of the following expressions for an resourceId property

### ISSUE THREAD

**Issue by marrobi** [Azure/arm-ttk#64]
In app gateway keyVaultSecretId takes the following format:

```
"sslCertificates": [
                    {
                        "name": "mycert",
                        "properties": {
                            "keyVaultSecretId": "[concat(reference(resourceid('Microsoft.KeyVault/vaults',parameters('keyVaultName')),'2019-09-01').vaultUri,'secrets/mycert')]"
                        }
                    }
                ],
```

Not sure I can do this with one of the resource ID functions?

Thanks.

**Comment by bmoore-msft** [comment]
hold one... wrong test

**Comment by bmoore-msft** [comment]
ok fixed in #68
